argus/backend/service/planner_service.py

```diff
@@ -438,3 +448,88 @@ def resolve_plan(self, plan_id: str | UUID) -> list[dict[str, Any]]:
                 ent["group"] = group.pretty_name or group.name
 
         return mapped
+    
+    def trigger_jobs(self, payload: PlanTriggerPayload) -> bool:
+
+        release_name = payload.get("release")
+        plan_id = payload.get("plan_id")
+        version = payload.get("version")
+
+        condition_set = (bool(release_name), bool(plan_id), bool(version))
+
+        match condition_set:
+            case (True, False, False):
+                release = ArgusRelease.get(name=release_name)
+                filter_expr = { "release_id__eq": release.id }
+            case (False, True, False):
+                filter_expr = { "id__eq": plan_id }
+            case (False, False, True):
+                filter_expr = { "target_version__eq": version }
+            case _:
+                raise PlannerServiceException("No version, release name or plan id specified.", payload)
+
+        LOGGER.info("Filter %s", filter_expr)
+        plans: list[ArgusReleasePlan] = list(ArgusReleasePlan.filter(**filter_expr).all())
+
+        if len(plans) == 0:
+            return False, "No plans to trigger"
+
+        common_params = payload.get("common_params", {})
+        params = payload.get("params", [])
+        test_ids = [test_id for plan in plans for test_id in plan.tests]
+        group_ids = [group_id for plan in plans for group_id in plan.groups]
+
+        tests = []
+        for batch in chunk(test_ids):
+            tests.extend(ArgusTest.filter(id__in=batch).all())
+
+        for batch in (chunk(group_ids)):
+            tests.extend(ArgusTest.filter(group_id__in=batch).allow_filtering().all())
+
+        LOGGER.info("Will trigger %s tests...", len(tests))
+
+        service = JenkinsService()
+        failures = []
+        successes = []
+        for test in tests:
+            try:
+                latest_build_number = service.latest_build(test.build_system_id)
+                if latest_build_number == -1:
+                    failures.append(test.build_system_id)
+                    continue
```

Comment:
In case we have fresh release, won't jobs be without first runs?
how about triggering it in that case, wait and try again?

Reply:
So in theory SCT starts the first build when it syncs the release tree, I could also try firing off a job without parameters to just get it past `#1` and executed on next call.